components/brave_wallet_ui/page/screens/composer_ui/select_token_modal/select_token_modal.tsx

```diff
@@ -80,31 +88,95 @@ import {
 import {
   VirtualizedTokenList //
 } from '../virtualized_token_list/virtualized_tokens_list'
+import { SelectAccount } from './select_account/select_account'
+import { TokenDetails } from './token_details/token_details'
+import { NetworksDropdown } from '../dropdowns/networks_dropdown'
+import { AccountsDropdown } from '../dropdowns/accounts_dropdown'
+import {
+  BottomSheet //
+} from '../../../../components/shared/bottom_sheet/bottom_sheet'
 
 // Styled Components
 import { Row, Column } from '../../../../components/shared/style'
 import {
   ScrollContainer,
-  AccountSection,
-  AccountAddressText,
-  AccountNameText,
-  BalanceText,
-  SearchBarRow,
   NoAssetsText,
-  SendOptionsRow
+  SearchInput
 } from './select_token_modal.style'
 
-const getIsTokenSelected = (
-  token: BraveWallet.BlockchainToken,
-  selectedToken?: BraveWallet.BlockchainToken
+const checkIsAccountOptionDisabled = (
+  account: BraveWallet.AccountInfo,
+  network: BraveWallet.NetworkInfo
+) => {
+  if (
+    network.chainId === AllNetworksOption.chainId ||
+    account.address === AllAccountsOption.address
+  ) {
+    return false
+  }
+  return network.coin !== account.accountId.coin
+}
+
+const checkIsNetworkOptionDisabled = (
+  network: BraveWallet.NetworkInfo,
+  account: BraveWallet.AccountInfo
 ) => {
-  if (!selectedToken) {
+  if (
+    account.address === AllAccountsOption.address ||
+    network.chainId === AllNetworksOption.chainId
+  ) {
     return false
   }
-  return (
-    selectedToken.contractAddress === token.contractAddress &&
-    selectedToken.coin === token.coin &&
-    selectedToken.chainId === token.chainId
+  return account.accountId.coin !== network.coin
+}
+
+const getFullAssetBalance = (
+  asset: BraveWallet.BlockchainToken,
+  networks: BraveWallet.NetworkInfo[],
+  accounts: BraveWallet.AccountInfo[],
+  tokenBalancesRegistry?: TokenBalancesRegistry | null
+) => {
+  if (!tokenBalancesRegistry) {
+    return ''
+  }
+
+  const network = networks.find(
+    (network) =>
+      network.coin === asset.coin && network.chainId === asset.chainId
+  )
+
+  const amounts = accounts
+    .filter((account) => {
+      return network && networkSupportsAccount(network, account.accountId)
+    })
+    .map((account) =>
+      getBalance(account.accountId, asset, tokenBalancesRegistry)
+    )
```

Comment:
Can we return early if no account was found?

```suggestion
  const network = networks.find(
    (network) =>
      network.coin === asset.coin && network.chainId === asset.chainId
  )
  
  if (!network) {
    return "0"
  }

  const amounts = accounts
    .filter((account) => {
      return networkSupportsAccount(network, account.accountId)
    })
    .map((account) =>
      getBalance(account.accountId, asset, tokenBalancesRegistry)
    )
```

Reply:
Will now return early if no `network` is found